### ISSUE TITLE
Accessibility: Login form missing required indication for screen reader users

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43658]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

See: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/templates/global/form-login.php

The label for the field is read as "Username or email address star" and "password star." The inputs are missing the required attribute.

### Expected behavior

Add an aria-label to the asterisk or change it to visible required text (preferred).
A) Best:
`<label for="username"><?php esc_html_e( 'Username or email', 'woocommerce' ); ?>&nbsp;<span class="required">(required)</span></label>`

B) Sufficient:
`<label for="username"><?php esc_html_e( 'Username or email', 'woocommerce' ); ?>&nbsp;<span class="required" aria-label="required">*</span></label>`

Add the required attribute to the inputs like this:
` <input type="text" class="input-text" name="username" id="username" autocomplete="username" required />`

### Actual behavior

No required indicator for screen reader users.

### Steps to reproduce

See linked code above.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
